spec/models/atmosphere/user_fund_spec.rb

```diff
@@ -33,4 +33,75 @@
         to include(I18n.t('funds.unique_user'))
   end
 
+
+  describe '#ensure_single_default' do
+
+    it 'makes first user_fund default' do
+      expect(user_fund.default).to be_truthy
+    end
+
+    it 'blocks setting the only fund as not default' do
+      expect(user_fund.default).to be_truthy
+      user_fund.default = false
+      user_fund.save
+      expect(user_fund.reload.default).to be_truthy
+    end
+
+    it 'does not bother single default fund creation' do
+      pc = create(:poor_chap)
+      uf = create(:user_fund, user: pc, default: true)
+      expect(uf.reload.default).to be_truthy
+    end
+
+    it 'keeps only single default user fund' do
+      uf = create(:user_fund, user: user_fund.user, default: true)
+      expect(Atmosphere::UserFund.count).to eq 2
+      expect(uf.reload.default).to be_truthy
+      expect(user_fund.reload.default).to be_falsey
+    end
+
+    it 'does not prematurely default to an invalid fund' do
+      begin
+        create(:user_fund,
+               fund: user_fund.fund,
+               user: user_fund.user,
+               default: true)
+      rescue ActiveRecord::RecordInvalid
```

Comment:
This can be changed into following construction:

```ruby
expect do
  create(:user_fund, ...)
end.to raise_error(ActiveRecord::RecordInvalid)
```

or (maybe even better)

```ruby
expect do
  create(:user_fund, ...)
end.to change { Atmosphere::UserFund.count }.by 0
```